### PR TITLE
Added Typescript definitions

### PR DESCRIPTION
**Pull request by dylanparry**
Hi,

I’ve just started using your library for testing my async action creators and thought it would be helpful to write a Typescript definition file for it. It’s a first attempt, so it’s possibly a little rough, but it seems to do the job for me so far. No doubt it can be improved upon, but it’s nice to have some definitions available at least :)

### CODE REVIEW

**Comment by arnaudbenard** [comment]
I'm all up for more clarity with the documentation, we can achieve the same thing with adding the types in the readme. What is the benefit of this?

**Comment by dylanparry** [comment]
It means that anyone choosing to develop in Typescript can import the type definitions using "typings" (https://www.npmjs.com/package/typings) and get context-sensitive tooltips etc in the IDE of their choice. Without a d.ts file any code using an external library shows up with red squiggles as the IDE is unable to determine what the "types" are for the parameters for each function (or that the functions even exist as part of the library).

untitled-1

Notice the helpful menu that appears in the screenshot? That’s only possible because of the d.ts file. My cursor is also hovering over the "dispatch" method in the menu, which then shows the tooltip to the right of it explaining how the function works, what parameters it needs and what it'll return.

Having the types as part of the documentation isn't quite enough for using a library in Typescript, but it’s a start. I noticed you already had good documentation for the library, which is what I used to help me create this d.ts file.

So to sum it up, it’s super helpful to have a d.ts file for any external library! React/Redux have them already, so having them in this library would be helpful too.

**Comment by arnaudbenard** [comment]
@dylanparry You sold it to me. Let's give it a try 👍
